pkg/compact/compact.go

```diff
@@ -27,28 +28,96 @@ import (
 // It sorts them into compaction groups based on equal label sets.
 type Syncer struct {
 	logger    log.Logger
+	reg       *prometheus.Registry
```

Comment:
`prometheus.Registerer` to avoid the weird confusion I just head in the other PR :)